lib/LocalServer.js

```diff
@@ -25,43 +26,24 @@ LocalServer.prototype.createSocketListener = function() {
 
     var me  = this;
 
+    var routes = {
+        'log': me.onDeviceLog,
+        'disconnect': me.onTestsCompletedOrDisconnected,
+        'jasmineStarted': specReporters.jasmineStarted,
+        'specStarted': specReporters.specStarted,
+        'specDone': specReporters.specDone,
+        'suiteStarted': specReporters.suiteStarted, 
+        'suiteDone': specReporters.suiteDone,
+        'jasmineDone': me.onJasmineDone
+    };
+
     listener.on('connection', function(socket) {
         logger.info('local-server: new socket connection');
         me.connection = socket;
 
-        socket.on('log', me.onDeviceLog);
-
-        socket.on('disconnect', function() {
-            me.onTestsCompletedOrDisconnected();
-        });
-
-        socket.on('jasmineStarted', function(data) {
-            specReporters.jasmineStarted(data);
-        });
-
-        socket.on('specStarted', function(data) {
-            specReporters.specStarted(data);
-        });
-
-        socket.on('specDone', function(data) {
-            specReporters.specDone(data);
-        });
-
-        socket.on('suiteStarted', function(data) {
-            specReporters.suiteStarted(data);
-        });
-
-        socket.on('suiteDone', function(data) {
-            specReporters.suiteDone(data);
-        });
-
-        socket.on('jasmineDone', function(data) {
-            specReporters.jasmineDone(data);
-            // save results to report them later
-            me.onTestsCompleted();
-            // disconnect because all tests have been completed
-            socket.disconnect();
-        });
+        for (var routeType in routes) {
+            socket.on(routeType, routes[routeType].bind(me));
```

Comment:
This will bind `specReporters`  calls to LocalServer instances as well, should we bind when defining routes instead?